include/NeoFOAM/core/dictionary.hpp

```diff
@@ -95,7 +95,7 @@ class Dictionary
         catch (const std::bad_any_cast& e)
         {
             logBadAnyCast<T>(e, key, data_);
-            throw e;
+            throw;
```

Comment:
```suggestion
            throw e;
```
accidentally deleted?

Reply:
throw without parameter rethrows the exception currently being handled